braindecode/datautil/windowers.py

```diff
@@ -190,15 +208,20 @@ def create_fixed_length_windows(
     _check_windowing_arguments(
         start_offset_samples, stop_offset_samples,
         window_size_samples, window_stride_samples)
+    if stop_offset_samples == 0:
+        warnings.warn(
+            'Meaning of `trial_stop_offset_samples`=0 has changed, use `None` '
+            'to indicate end of trial/recording. Using `None`.')
+        stop_offset_samples = None
 
     list_of_windows_ds = []
     for ds in concat_ds.datasets:
-        stop = ds.raw.n_times if stop_offset_samples == 0 else stop_offset_samples
-        stop = stop - window_size_samples
+        stop = ds.raw.n_times \
+            if stop_offset_samples is None else stop_offset_samples
+        stop = stop - window_size_samples + ds.raw.first_samp
```

Comment:
yeah great, not even sure what was logic behind previous decision. seems quite strange. I wonder actually if we had intended stop offset samples to refer to offset to end of recording? as it is in from events thingy and end of trial?

Reply:
Aah you have a good point, that's probably what was going on. I think the new approach might be more useful: let's say you want to extract windows from 10 minutes of a recording - you can provide the onset and offset as 2 and 12 mins, which is more natural than 2 and *calculate length of recording* - (2 + 10). What do you think?

Reply:
I agree that extracting a particular length if you don't have events might be what you want in many cases and would also keep like this for this PR. Only question long-term is if we want to have both options (stop given from beginning of recording and from end of recording), one could also do it in every dirty way: if negative, assume it is from end of recording, if positive, assume from start of recording... but as said let's keep in this way (from start of recording) for this PR